File: src/HelloWorldVB.UnitTests/DebuggerDisplayClassTest.vb
Imports System.Reflection
Imports HelloWorld
Imports NUnit.Framework

<TestFixture>
Public Class DebuggerDisplayClassTest

	<Test>
	Public Sub DebuggerDisplay_StringMethod_String17()

		Dim cls As New DebuggerDisplayClassStringMethod With
		{
			.Number = 17
		}
		Dim dd As MethodInfo = cls.GetType().GetMethod("DebuggerDisplay", BindingFlags.Instance Or BindingFlags.NonPublic)

		Dim act As Object = dd.Invoke(cls, Nothing)
		Dim exp As Object = "17"

		Assert.AreEqual(exp, act)
	End Sub

	<Test>
	Public Sub DebuggerDisplay_ObjectMethod_Int17()

		Dim cls As New DebuggerDisplayClassObjectMethod With
		{
			.Number = 17
		}

		Dim dd As MethodInfo = cls.GetType().GetMethod("DebuggerDisplay", BindingFlags.Instance Or BindingFlags.NonPublic)

		Dim act As Object = dd.Invoke(cls, Nothing)
		Dim exp = 17

		Assert.AreEqual(exp, act)
	End Sub

	<Test>
	Public Sub DebuggerDisplay_StringProperty_String17()

		Dim cls As New DebuggerDisplayClassStringProperty With
		{
			.Number = 17
		}
		Dim dd As PropertyInfo = cls.GetType().GetProperty("DebuggerDisplay", BindingFlags.Instance Or BindingFlags.NonPublic)

		Dim act As Object = dd.GetValue(cls)
		Dim exp As Object = "17"

		Assert.AreEqual(exp, act)
	End Sub

	<Test>
	Public Sub DebuggerDisplay_ObjectProperty_Int17()

		Dim cls As New DebuggerDisplayClassObjectProperty With
		{
			.Number = 17
		}

		Dim dd As PropertyInfo = cls.GetType().GetProperty("DebuggerDisplay", BindingFlags.Instance Or BindingFlags.NonPublic)

		Dim act As Object = dd.GetValue(cls)
		Dim exp As Object = 17

		Assert.AreEqual(exp, act)
	End Sub

End Class

File: test/HelloWorld.UnitTests/DebuggerDisplayClassTest.cs
using NUnit.Framework;
using System.Reflection;

namespace HelloWorld.UnitTests
{
	[TestFixture]
	public class DebuggerDisplayClassTest
	{
		[Test]
		public void DebuggerDisplay_StringMethod_String17()
		{
			var cls = new DebuggerDisplayClassStringMethod() { Number = 17 };
			var dd = cls.GetType().GetMethod("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic);

			var act = dd.Invoke(cls, new object[0]);
			var exp = "17";

			Assert.AreEqual(exp, act);
		}

		[Test]
		public void DebuggerDisplay_ObjectMethod_Int17()
		{
			var cls = new DebuggerDisplayClassObjectMethod() { Number = 17 };
			var dd = cls.GetType().GetMethod("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic);

			var act = dd.Invoke(cls, new object[0]);
			var exp = 17;

			Assert.AreEqual(exp, act);
		}

		[Test]
		public void DebuggerDisplay_StringProperty_String17()
		{
			var cls = new DebuggerDisplayClassStringProperty() { Number = 17 };
			var dd = cls.GetType().GetProperty("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic);

			var act = dd.GetValue(cls);
			var exp = "17";

			Assert.AreEqual(exp, act);
		}

		[Test]
		public void DebuggerDisplay_ObjectProperty_Int17()
		{
			var cls = new DebuggerDisplayClassObjectProperty() { Number = 17 };
			var dd = cls.GetType().GetProperty("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic);

			var act = dd.GetValue(cls);
			var exp = 17;

			Assert.AreEqual(exp, act);
		}
	}
}
